### PR TITLE
Actually remove all status validations

### PR DESCRIPTION
**Pull request by matthchr**

### CODE REVIEW

File: v2/tools/generator/internal/codegen/pipeline/remove_status_property_validations.go [hunk 1]
```diff
@@ -114,3 +115,12 @@ func removeValidatedType(this *astmodel.TypeVisitor, vt *astmodel.ValidatedType,
 func removeEnumValidations(this *astmodel.TypeVisitor, et *astmodel.EnumType, _ interface{}) (astmodel.Type, error) {
 	return et.WithoutValidation(), nil
 }
+
+// removeKubebuilderRequired removes kubebuilder:validation:Required from all properties
+func removeKubebuilderRequired(this *astmodel.TypeVisitor, ot *astmodel.ObjectType, ctx interface{}) (astmodel.Type, error) {
```
Comment (theunrepentantgeek) [inline v2/tools/generator/internal/codegen/pipeline/remove_status_property_validations.go:120]: Isn't this going to remove it from ALL `ObjectType` instances - don't you only want to do this for Status types?

**Review by theunrepentantgeek — CHANGES_REQUESTED** [review]
Mostly LGTM, but I have a question about blast radius.

File: v2/tools/generator/internal/codegen/pipeline/remove_status_property_validations.go (hunk 1 above)
Comment (matthchr) [reply v2/tools/generator/internal/codegen/pipeline/remove_status_property_validations.go:120]: If you look up in this file on line 57, we only run this on status types.

Comment (theunrepentantgeek) [reply v2/tools/generator/internal/codegen/pipeline/remove_status_property_validations.go:120]: All good then. 😁 #approved

**Review by theunrepentantgeek — APPROVED** [review]

**Pull request merged**